src/sync/treiber_stack.rs

```diff
@@ -1,82 +1,107 @@
+//! Treiber stacks.
+
 use std::sync::atomic::Ordering::{Acquire, Release, Relaxed};
 use std::ptr;
 
-use mem::epoch::{self, Atomic, Owned};
+use epoch::{self, Atomic};
 
 /// Treiber's lock-free stack.
 ///
-/// Usable with any number of producers and consumers.
+/// This is usable with any number of producers and consumers.
 #[derive(Debug)]
 pub struct TreiberStack<T> {
+    /// The head (top) node.
     head: Atomic<Node<T>>,
 }
 
+/// A node in the stack.
 #[derive(Debug)]
 struct Node<T> {
+    /// The payload of this node.
     data: T,
+    /// The node below this node.
     next: Atomic<Node<T>>,
 }
 
 impl<T> TreiberStack<T> {
     /// Create a new, empty stack.
     pub fn new() -> TreiberStack<T> {
-        TreiberStack { head: Atomic::null() }
+        TreiberStack {
+            head: Atomic::null(),
+        }
     }
 
-    /// Push `t` on top of the stack.
-    pub fn push(&self, t: T) {
-        let mut n = Owned::new(Node {
-            data: t,
+    /// Push an element on top of the stack.
+    pub fn push(&self, elem: T) {
+        // Construct the node.
```

Comment:
Overall I'd refrain from these comments, unless the comment is badly worded or incorrect. The code review surface is already huge to start nitpicking these.